pkg/3scale/amp/operator/system_options_provider.go

```diff
@@ -559,3 +561,30 @@ func (s *SystemOptionsProvider) sphinxPodTemplateLabels() map[string]string {
 
 	return labels
 }
+
+func (s *SystemOptionsProvider) isSTS() (bool, error) {
+	if s.apimanager.Spec.System == nil ||
+		s.apimanager.Spec.System.FileStorageSpec == nil ||
+		s.apimanager.Spec.System.FileStorageSpec.S3 == nil {
+		return false, nil
+	}
+	awsCredentialsSecretName := s.apimanager.Spec.System.FileStorageSpec.S3.ConfigurationSecretRef.Name
+	if awsCredentialsSecretName == "" {
```

Comment:
```suggestion
if s.apimanager.Spec.System == nil ||
		s.apimanager.Spec.System.FileStorageSpec == nil ||
		s.apimanager.Spec.System.FileStorageSpec.S3 == nil ||
                 s.apimanager.Spec.System.FileStorageSpec.S3.ConfigurationSecretRef.Name != "" {
```

Reply:
or maybe we should return an error if the values nil since we expect them to be put there when the system provider is set up.

Reply:
thank you for comment.  Actually we are checking ConfigurationSecretRef.Name a two lines later (572, 573), and providing Info  ... 
But yes, this notification seems not exact:  	
if awsCredentialsSecretName == "" {
		return false, fmt.Errorf("no aws credentials provided")
	}